src/commands/runFromPackageDeploy.ts

```diff
@@ -0,0 +1,212 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as archiver from 'archiver';
+// tslint:disable-next-line:no-require-imports
+import StorageClient = require('azure-arm-storage');
+import { StorageAccountListKeysResult } from 'azure-arm-storage/lib/models';
+import { StringDictionary } from 'azure-arm-website/lib/models';
+import * as azureStorage from "azure-storage";
+import * as fs from 'fs';
+import * as fse from 'fs-extra';
+import * as os from 'os';
+import * as path from 'path';
+import { MessageItem, Progress, ProgressLocation, window, workspace } from "vscode";
+import { SiteClient } from 'vscode-azureappservice';
+import { DialogResponses, IActionContext, IAzureParentNode } from 'vscode-azureextensionui';
+import { localSettingsFileName, ProjectLanguage } from '../constants';
+import { ArgumentError } from '../errors';
+import { ext } from '../extensionVariables';
+import { localize } from '../localize';
+import { getFuncExtensionSetting, getProjectLanguage } from '../ProjectSettings';
+import { updateWorkspaceSetting } from '../ProjectSettings';
+import { FunctionAppTreeItem } from '../tree/FunctionAppTreeItem';
+import * as  azUtil from '../utils/azure';
+import * as fsUtil from '../utils/fs';
+
+/**
+ * Method of deployment that is only intended to be used for Linux Consumption Function apps.
+ * To deploy with Run from Package on a Windows plan, create the app setting "WEBSITE_RUN_FROM_ZIP" and set it to "1".
+ * Then deploy via "zipdeploy" as usual.
+ */
+
+export async function runFromPackageDeploy(actionContext: IActionContext, node: IAzureParentNode<FunctionAppTreeItem>, fsPath: string): Promise<void> {
+    const language: ProjectLanguage = await getProjectLanguage(fsPath, ext.ui);
```

Comment:
no need to get language again. Just pass it in as a parameter to runFromPackageDeploy